tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem_test.cc

```diff
@@ -12,50 +12,150 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+#include <random>
+
+#include "google/cloud/storage/client.h"
 #include "tensorflow/c/experimental/filesystem/filesystem_interface.h"
 #include "tensorflow/c/tf_status_helper.h"
+#include "tensorflow/core/platform/path.h"
 #include "tensorflow/core/platform/stacktrace_handler.h"
 #include "tensorflow/core/platform/test.h"
 
-#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x))
+#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x)) << TF_Message(x)
+
+static const char* content = "abcdefghijklmnopqrstuvwxyz1234567890";  // 36
+
+namespace gcs = google::cloud::storage;
 
 // Forward declaration
+void ParseGCSPath(absl::string_view fname, bool object_empty_ok, char** bucket,
+                  char** object, TF_Status* status);
+
+namespace tf_random_access_file {
+void Cleanup(TF_RandomAccessFile* file);
+int64_t Read(const TF_RandomAccessFile* file, uint64_t offset, size_t n,
+             char* buffer, TF_Status* status);
+}  // namespace tf_random_access_file
+
 namespace tf_gcs_filesystem {
 void Init(TF_Filesystem* filesystem, TF_Status* status);
-}
+void Cleanup(TF_Filesystem* filesystem);
+void NewRandomAccessFile(const TF_Filesystem* filesystem, const char* path,
+                         TF_RandomAccessFile* file, TF_Status* status);
+}  // namespace tf_gcs_filesystem
 
 namespace tensorflow {
 namespace {
 
 class GCSFilesystemTest : public ::testing::Test {
  public:
   void SetUp() override {
+    root_dir_ = io::JoinPath(
+        tmp_dir_,
+        ::testing::UnitTest::GetInstance()->current_test_info()->name());
     status_ = TF_NewStatus();
     filesystem_ = new TF_Filesystem;
     tf_gcs_filesystem::Init(filesystem_, status_);
-    ASSERT_TF_OK(status_) << "Can not initialize filesystem. "
+    ASSERT_TF_OK(status_) << "Could not initialize filesystem. "
                           << TF_Message(status_);
   }
   void TearDown() override {
     TF_DeleteStatus(status_);
-    // TODO(vnvo2409): Add filesystem cleanup
+    tf_gcs_filesystem::Cleanup(filesystem_);
     delete filesystem_;
   }
 
+  static bool InitializeTmpDir() {
+    // This env should be something like `gs://bucket/path`
```

Comment:
Can we test that the env variable has this format before continuing?

Reply:
I will pass the env to ParseGCSPath